airflow/providers/cncf/kubernetes/kubernetes_helper_functions.py

```diff
@@ -43,8 +43,14 @@ def rand_str(num):
     return "".join(secrets.choice(alphanum_lower) for _ in range(num))
 
 
-def add_pod_suffix(*, pod_name, rand_len=8, max_len=80):
-    """Add random string to pod name while staying under max len."""
+def add_pod_suffix(pod_name: str, rand_len: int = 8, max_len: int = 80) -> str:
```

Comment:
```suggestion
def add_pod_suffix(*, pod_name: str, rand_len: int = 8, max_len: int = 80) -> str:
```

Why'd we drop the kwargs only behavior?

Reply:
MyPy had problems with it.

Reply:
I added it back. When I did it, MyPy started to report another error which made completely no sense:

<img width="577" alt="Screenshot 2023-07-25 at 21 46 30" src="https://github.com/apache/airflow/assets/595491/8dd7a9e3-94b0-4a08-89fb-b8e1e6340c53">

So I had to type-ignore it. But it seems to work after that.